### ISSUE TITLE
Unable to register CRDs

### ISSUE THREAD

**Issue by nzhuang123** [cilium/cilium#17462]
When deploying cilium on Suse linux 15, got the following error msg, and the cilium daemon pod is crashing:

level=fatal msg="Unable to register CRDs" error="Unable to create custom resource definition: the server could not find the requested resource" subsys=daemon

suse-1:/pkgs/sfnc/vm-files/sfnc-1.0.0/lcm # kubectl get pods -n kube-system
NAME                               READY   STATUS              RESTARTS      AGE
cilium-base-t7xj8                  0/1     CrashLoopBackOff    8 (58s ago)   17m
cilium-operator-869f7dc84b-sbs9h   1/1     Running             3 (83s ago)   17m
coredns-78fcd69978-j844q           0/1     ContainerCreating   0             19h
coredns-78fcd69978-mhqvt           0/1     ContainerCreating   0             19h
etcd-suse-1                        1/1     Running             0             19h
kube-apiserver-suse-1              1/1     Running             0             19h
kube-controller-manager-suse-1     1/1     Running             0             19h
kube-proxy-p5x7f                   1/1     Running             0             19h
kube-scheduler-suse-1              1/1     Running             0             19h

Can someone help to point out what might be wrong?

Thanks,

**Comment by pchaigno** [comment]
What's the state of the cilium-operator pods? Could you join a Cilium sysdump as per the issue template?

**Comment by nzhuang123** [comment]
cilium operator pod seems to be running fine:

cilium-operator-869f7dc84b-sbs9h   1/1     Running             3 (83s ago)   17m

I'm new to cilium, so it could be I'm missing some of the config...  How to join a cilium sysdump?

**Comment by pchaigno** [comment]
See the template you deleted: https://github.com/cilium/cilium/blob/master/.github/ISSUE_TEMPLATE/bug_template.md.

**Comment by nzhuang123** [comment]
**cilium version**
Client: 1.8.5  go version go1.13.15 linux/amd64
Daemon: Not responding

**uname -a**
Linux suse-1 5.3.18-57-default #1 SMP Wed Apr 28 10:54:41 UTC 2021 (ba3c2e9/lp-5d9e8aa) x86_64 x86_64 x86_64 GNU/Linux

**kubectl version**
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:38:50Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}

[cilium-sysdump-20210922-213806.zip](https://github.com/cilium/cilium/files/7219850/cilium-sysdump-20210922-213806.zip)

**Comment by pchaigno** [comment]
Your cilium-operator pod is failing with:
```
2021-09-23T04:32:51.374487910Z level=error msg=k8sError error="github.com/cilium/cilium/operator/watchers/cilium_endpoint.go:83: Failed to list *v2.CiliumEndpoint: the server could not find the requested resource (get ciliumendpoints.cilium.io)" subsys=k8s
2021-09-23T04:33:04.165600614Z level=fatal msg="Unable to start managing Kubernetes identities" error="timeout waiting for CRD ciliumidentities.cilium.io: timed out waiting for the condition" subsys=cilium-operator-generic
```
Are you sure it's able to connect to kube-apiserver?

**Comment by nzhuang123** [comment]
From the log, looks like it's connected to the apiserver.... are there any other command to test this connection?

level=info msg="Initializing daemon" subsys=daemon
**level=info msg="Establishing connection to apiserver" host="https://172.18.1.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s**
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=100.94.72.101 mtu=1500 subsys=mtu
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"enable-host-reachable-services\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=info msg="Restored services from maps" failed=0 restored=0 subsys=service
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="No old endpoints found." subsys=daemon
level=error msg="Command execution failed" cmd="[iptables -t mangle -n -L CILIUM_PRE_mangle]" error="exit status 1" subsys=iptables
level=warning msg="iptables: No chain/target/match by that name." subsys=iptables
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumNode subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumEndpoint subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumIdentity subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
level=fatal msg="Unable to register CRDs" error="Unable to create custom resource definition: the server could not find the requested resource" subsys=daemon

**Comment by vincentmli** [comment]
> From the log, looks like it's connected to the apiserver.... are there any other command to test this connection?
> 
> level=info msg="Initializing daemon" subsys=daemon
> **level=info msg="Establishing connection to apiserver" host="https://172.18.1.1:443" subsys=k8s level=info msg="Connected to apiserver" subsys=k8s**
> level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=100.94.72.101 mtu=1500 subsys=mtu
> level=info msg="Trying to auto-enable "enable-node-port", "enable-external-ips", "enable-host-reachable-services", "enable-host-port", "enable-session-affinity" features" subsys=daemon
> level=info msg="Restored services from maps" failed=0 restored=0 subsys=service
> level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
> level=info msg="Reading old endpoints..." subsys=daemon
> level=info msg="No old endpoints found." subsys=daemon
> level=error msg="Command execution failed" cmd="[iptables -t mangle -n -L CILIUM_PRE_mangle]" error="exit status 1" subsys=iptables
> level=warning msg="iptables: No chain/target/match by that name." subsys=iptables

@nzhuang123 I may have seen this error before but my memory not be right, it could be you have an old userland iptable installations that not working properly with kernel version for your distribution or the other way around, maybe something related to nftables

**Comment by nzhuang123** [comment]
Thanks for the reply. Any suggestion how to fix/work around this issue?

**Comment by vincentmli** [comment]
for me I was running centos 8 with custom built old kernel 5.0 if I recall correct, after I ran a newer kernel on centos 8, the problem went away, sorry i don't remember the detail and has not dug deep into it. you may check with suse community/vendor on how to workaround this

**Comment by vincentmli** [comment]
here is mine which is not exact same like   your error on centos 8 with 5.0 kernel, the log coming from cilium agent, not cilium operator, and no problem if I ran kernel 5.2 with centos 8

```
level=warning msg="-P PREROUTING ACCEPT" subsys=iptables
level=warning msg="-P INPUT ACCEPT" subsys=iptables
level=warning msg="-P FORWARD ACCEPT" subsys=iptables
level=warning msg="-P OUTPUT ACCEPT" subsys=iptables
level=warning msg="-P POSTROUTING ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="# Warning: ip6tables-legacy tables present, use ip6tables-legacy to see them" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-P PREROUTING ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-P OUTPUT ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="# Warning: ip6tables-legacy tables present, use ip6tables-legacy to see them" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-P INPUT ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-P FORWARD ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-P OUTPUT ACCEPT" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-N KUBE-FIREWALL" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-A INPUT -j KUBE-FIREWALL" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-A OUTPUT -j KUBE-FIREWALL" subsys=iptables
level=debug msg="Considering removing &{ip6tables} rule" obj="-A KUBE-FIREWALL -m comment --comment \"kubernetes firewall for dropping marked packets\" -m mark --mark 0x8000/0x8000 -j DROP" subsys=iptables
level=error msg="Command execution failed" cmd="[iptables -w 5 -t mangle -N CILIUM_POST_mangle]" error="exit status 4" subsys=iptables
level=warning msg="iptables v1.8.4 (nf_tables):  CHAIN_ADD failed (No such file or directory): chain OUTPUT" subsys=iptables
level=warning msg="Cannot list rules in table. Layer 7 proxy port may get reallocated, which could cause disruption for traffic selected by L7 policyiptablesmangle" error="exit status 4" prog=iptables subsys=iptables table=mangle
level=fatal msg="Error while creating daemon" error="error while initializing daemon: cannot add custom chain CILIUM_POST_mangle: exit status 4" subsys=daemon. 

```

**Comment by vincentmli** [comment]
I wonder if you turn on cilium agent debug log, if the cilium agent log would log similar error like mine, since my cilium agent has debug turned on. see https://docs.cilium.io/en/stable/cmdref/cilium-agent/ argument to turn on debug, it can be set in manifest or helm.

**Comment by nzhuang123** [comment]
For me the issue is CRD cannot be found, and the cilium pod is crashing..

level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.8.4 17d623853 2020-09-30T17:31:44-07:00 go version go1.14.9 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: 1177896bebde79915fe5f9092409bf0254084b4e/1.14.5/Modified/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.3.18) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Valid label prefix configuration:" subsys=labels-filter
level=info msg=" - :io.kubernetes.pod.namespace" subsys=labels-filter
level=info msg=" - :io.cilium.k8s.namespace.labels" subsys=labels-filter
level=info msg=" - :app.kubernetes.io" subsys=labels-filter
level=info msg=" - !:io.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes.io" subsys=labels-filter
level=info msg=" - !:.*beta.kubernetes.io" subsys=labels-filter
level=info msg=" - !:k8s.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.101.0.0/16
level=info msg="Using autogenerated IPv6 allocation range" subsys=node v6Prefix="f00d::a65:0:0:0/96"
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://172.18.1.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=100.94.72.101 mtu=1500 subsys=mtu
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"enable-host-reachable-services\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=info msg="Restored services from maps" failed=0 restored=0 subsys=service
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="No old endpoints found." subsys=daemon
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=error msg="Command execution failed" cmd="[iptables -t mangle -n -L CILIUM_PRE_mangle]" error="exit status 1" subsys=iptables
level=warning msg="iptables: No chain/target/match by that name." subsys=iptables
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumIdentity subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumEndpoint subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumNode subsys=k8s
**level=fatal msg="Unable to register CRDs" error="Unable to create custom resource definition: the server could not find the requested resource" subsys=daemon**

**Comment by vincentmli** [comment]
my cilium agent pod is crashing too, your log has no debug enabled, try to enable debug, you should see more debug log

**Comment by pchaigno** [comment]
> From the log, looks like it's connected to the apiserver.... are there any other command to test this connection?

@nzhuang123 That's the agent logs, not the cilium-operator logs.

@vincentmli I don't think it's the same issue. The iptables error in the present case seems likely to be a red herring whereas you had other, fairly different log lines in your case.

**Comment by vincentmli** [comment]
@pchaigno right, i am hoping @nzhuang123 could provide debug log, that may show more clue :)

**Comment by nzhuang123** [comment]
Could you tell me what log flag to enable?

**Comment by nzhuang123** [comment]
Also, in the operator log, I could see it's trying the list resource from v1beta1, but shouldn't it be v1 instead?

level=error msg=k8sError error="k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to list *v1beta1.CustomResourceDefinition: the server could not find the requested resource" subsys=k8s

**Comment by nzhuang123** [comment]
The following command also gives error:

kubectl get ciliumid
error: the server doesn't have a resource type "ciliumid"

**Comment by vincentmli** [comment]
change configmap cilium-config and set  `debug: "true"`
`kubectl edit configmap cilium-config -n kube-system ; kubectl delete pod -l k8s-app=cilium -A; `
you may also consider deploy most recent cilium version, 

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: cilium-config
  namespace: kube-system
data:


  # If you want to run cilium in debug mode change this value to true
  debug: "true"

```

**Comment by nzhuang123** [comment]
level=info msg="Cilium 1.8.4 17d623853 2020-09-30T17:31:44-07:00 go version go1.14.9 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: 1177896bebde79915fe5f9092409bf0254084b4e/1.14.5/Modified/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.3.18) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Valid label prefix configuration:" subsys=labels-filter
level=info msg=" - :io.kubernetes.pod.namespace" subsys=labels-filter
level=info msg=" - :io.cilium.k8s.namespace.labels" subsys=labels-filter
level=info msg=" - :app.kubernetes.io" subsys=labels-filter
level=info msg=" - !:io.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes.io" subsys=labels-filter
level=info msg=" - !:.*beta.kubernetes.io" subsys=labels-filter
level=info msg=" - !:k8s.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.101.0.0/16
level=info msg="Using autogenerated IPv6 allocation range" subsys=node v6Prefix="f00d::a65:0:0:0/96"
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://172.18.1.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=debug msg="Starting new controller" name=k8s-heartbeat subsys=controller uuid=7713acdc-1c48-11ec-ae0b-000c297a8fce
level=debug msg="Controller func execution time: 833ns" name=k8s-heartbeat subsys=controller uuid=7713acdc-1c48-11ec-ae0b-000c297a8fce
level=debug msg="Skipping Leases support fallback discovery" subsys=k8s
level=info msg="Inheriting MTU from external network interface" device=eth0 ipAddr=100.94.72.101 mtu=1500 subsys=mtu
level=debug msg="getting identity cache for identity allocator manager" subsys=identity-cache
level=debug msg="creating new EventQueue" name=repository-change-queue numBufferedEvents=100 subsys=eventqueue
level=debug msg="creating new EventQueue" name=repository-reaction-queue numBufferedEvents=100 subsys=eventqueue
level=debug msg="Performing regular background work" subsys=nodemanager syncInterval=1m0s
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"enable-host-reachable-services\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lxc subsys=bpf
level=debug msg="Opening map in parallel mode" file-path=/sys/fs/bpf/tc/globals/cilium_ipcache name=cilium_ipcache subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipcache subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_metrics subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_tunnel_map subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_reverse_sk subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_reverse_sk subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_sk subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_sk subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_services_v2 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_backends subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_reverse_nat subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_services_v2 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_backends subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_reverse_nat subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_events subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_events subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_signals subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_signals subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_call_policy subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_call_policy subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct4_global subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct4_global subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct_any4_global subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct_any4_global subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct6_global subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct6_global subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct_any6_global subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ct_any6_global subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_snat_v4_external subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_snat_v4_external subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_snat_v6_external subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_snat_v6_external subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh6 subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh6 subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipv4_frag_datagrams subsys=bpf
level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_ipv4_frag_datagrams subsys=bpf
level=debug msg="Starting new controller" name=metricsmap-bpf-prom-sync subsys=controller uuid=771504ba-1c48-11ec-ae0b-000c297a8fce
level=debug msg="Controller func execution time: 8.932µs" name=metricsmap-bpf-prom-sync subsys=controller uuid=771504ba-1c48-11ec-ae0b-000c297a8fce
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb_affinity_match subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_affinity subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_affinity subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb4_source_range subsys=bpf
level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_lb6_source_range subsys=bpf
level=info msg="Restored services from maps" failed=0 restored=0 subsys=service
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="No old endpoints found." subsys=daemon
level=debug msg="Starting new controller" name=dns-garbage-collector-job subsys=controller uuid=771544ad-1c48-11ec-ae0b-000c297a8fce
level=info msg="Envoy: Starting xDS gRPC server listening on /var/run/cilium/xds.sock" subsys=envoy-manager
level=debug msg="Controller func execution time: 3.357µs" name=dns-garbage-collector-job subsys=controller uuid=771544ad-1c48-11ec-ae0b-000c297a8fce
level=error msg="Command execution failed" cmd="[iptables -t mangle -n -L CILIUM_PRE_mangle]" error="exit status 1" subsys=iptables
level=warning msg="iptables: No chain/target/match by that name." subsys=iptables
level=debug msg="DNS Proxy port is configured to 0. A random port will be assigned by the OS." subsys=fqdn/dnsproxy
level=debug msg="DNS Proxy bound to address" address="[::]:38915" subsys=fqdn/dnsproxy
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumNode subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumEndpoint subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=v2.CiliumIdentity subsys=k8s
level=fatal msg="Unable to register CRDs" error="Unable to create custom resource definition: the server could not find the requested resource" subsys=daemon

**Comment by pchaigno** [comment]
Vincent is correct, Cilium v1.8.4 is very old. Please try with the latest v1.8.x at the very least.

**Comment by nzhuang123** [comment]
Running v1.10.4, and I'm getting this error:

level=info msg="Cilium Operator 1.10.4 2a46fd6 2021-09-01T12:58:41-07:00 go version go1.16.7 linux/amd64" subsys=cilium-operator-generic
level=info msg="Starting apiserver on address 127.0.0.1:9234" subsys=cilium-operator-api
level=info msg="Establishing connection to apiserver" host="https://172.18.1.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=debug msg="Starting new controller" name=k8s-heartbeat subsys=controller uuid=55070ee0-1249-4359-ad2c-bde1ecff32e5
level=debug msg="Controller func execution time: 7.472951ms" name=k8s-heartbeat subsys=controller uuid=55070ee0-1249-4359-ad2c-bde1ecff32e5
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumLocalRedirectPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumNode/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumEgressNATPolicy/v2alpha1 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumEndpoint/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumExternalWorkload/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumIdentity/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
level=info msg="Creating CRD (CustomResourceDefinition)..." name=CiliumNetworkPolicy/v2 subsys=k8s
**level=fatal msg="Unable to register CRDs" error="Unable to create custom resource definition: customresourcedefinitions.apiextensions.k8s.io is forbidden: User \"system:serviceaccount:kube-system:cilium-operator\" cannot create resource \"customresourcedefinitions\" in API group \"apiextensions.k8s.io\" at the cluster scope" subsys=cilium-operator-generic**

**Comment by nzhuang123** [comment]
By giving the create permission to the Operator pods, and with v1.10.4, it works! Thanks a lot!

**Comment by nzhuang123** [comment]
Just wondering why do we need the crd create permission for v1.10.4?  I'm using the same cilium.yaml on another ubuntu vm, and it works with just the get/list/watch permission

**Comment by vincentmli** [comment]
maybe different kubernetes version?

**Comment by pchaigno** [comment]
> Just wondering why do we need the crd create permission for v1.10.4?

I don't think that's new in v1.10. You need CRD Create permission to create Cilium's CRD.

> I'm using the same cilium.yaml on another ubuntu vm, and it works with just the get/list/watch permission

That's surprising. Maybe the CRDs were already installed?

Closing the issue as this appears to be solved.

**Comment by meysam81** [comment]
I had a similar issue, as mentioned in the title of this issue. The Cilium Daemonset was stuck and waiting for CRDs. After seeing [this comment](https://github.com/cilium/cilium/issues/17462#issuecomment-926023804), I checked my `kube-system` and it seemed like the old `deploy/cilium-operator` was still running, and that has direct conflict due to its affinity setting. For context, I did a `cilium upgrade` that got me here (from v1.13 to v1.15).
